test/dlc_tests/sanity/test_pre_release.py

```diff
@@ -141,6 +141,49 @@ def test_ubuntu_version(image):
 
 @pytest.mark.usefixtures("sagemaker")
 @pytest.mark.model("N/A")
+@pytest.mark.canary("Run non-gpu tf serving version test regularly on production images")
+def test_tf_serving_version_cpu(tensorflow_inference):
+    """
+    For non-huggingface non-GPU TF inference images, check that the tag version matches the version of TF serving
+    in the container.
+
+    Huggingface includes MMS and core TF, hence the versioning scheme is based off of the underlying tensorflow
+    framework version, rather than the TF serving version.
+
+    GPU inference images will be tested along side `test_framework_and_cuda_version_gpu` in order to be judicious
+    about GPU resources. This test can run directly on the host, and thus does not require additional resources
+    to be spun up.
+
+    @param tensorflow_inference: ECR image URI
+    """
+    # Set local variable to clarify contents of fixture
+    image = tensorflow_inference
+
+    if "gpu" in image:
+        pytest.skip(
+            "GPU images will have their framework version tested in test_framework_and_cuda_version_gpu")
+    if "neuron" in image:
+        pytest.skip(
+            "Neuron images will have their framework version tested in test_framework_and_neuron_sdk_version")
+
+    _, tag_framework_version = get_framework_and_version_from_tag(
+        image)
+
+    ctx = Context()
+    container_name = get_container_name("tf-serving-version", image)
+    start_container(container_name, image, ctx)
+    output = run_cmd_on_container(
+        container_name, ctx, "tensorflow_model_server --version", executable="bash"
+    )
+    assert (
+        re.match(rf"TensorFlow Model Server: {tag_framework_version}(\D+)?", output.stdout),
```

Comment:
I have noticed that sometimes the model server version is not intuitive. For instance, for TF2.7, it looks like this

"TensorFlow ModelServer: 2.7.0-rc1+dev.sha.no_git"

@hsl89 - is there a way we can reliably predict what this string will be? Why does the version sometimes have "rc" specified?

Reply:
@arjkesh , the version is determined by this file https://github.com/tensorflow/serving/blob/2.8.0/tensorflow_serving/model_servers/version.h
the entire string might not be predictable, because it depends on how many rc they have. I think a work around is to grep the major.minor.patch version

Reply:
@hsl89 is there a way to reliably predict which RC is the latest one, or ensure that the version of the released DLC is not built off of an RC?

I don't want to run into a situation where we have a binary with RC1, for instance, but the release version is built off of RC2